bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -29,211 +34,315 @@
  * @author Markus Rathgeb - Synchronize access to broker connections
  */
 public class MqttService implements ManagedService {
-
     private final Logger logger = LoggerFactory.getLogger(MqttService.class);
-
     private final Map<String, MqttBrokerConnection> brokerConnections = new ConcurrentHashMap<String, MqttBrokerConnection>();
-
+    private final List<MqttBrokersObserver> brokersObservers = new CopyOnWriteArrayList<>();
+    @Deprecated
     private EventPublisher eventPublisher;
 
-    @Override
-    public void updated(Dictionary<String, ?> properties) throws ConfigurationException {
-
-        // load broker configurations from configuration file
-        if (properties == null || properties.isEmpty()) {
-            return;
-        }
-
+    /**
+     * Transform the JDK 1.0 old-school, rotten, deprecated Dictionary content to something useful.
+     *
+     * @param properties An old-school, deprecated dictionary object with the service configuration.
+     * @return A 'list' of broker configurations as key-value maps. A configuration map at least contains a "name".
+     */
+    public Map<String, Map<String, Object>> extractBrokerConfigurations(Dictionary<String, ?> properties) {
+        Map<String, Map<String, Object>> configPerBroker = new TreeMap<String, Map<String, Object>>();
```

Comment:
The order is irrelevant. I will change it then. I was just concerned of the in average 13 unused entries in a hash map (with an initial default capacity of 16). But because we store string references only it might be just a few bytes per entry anyway.